aws/resource_aws_batch_compute_environment.go

```diff
@@ -0,0 +1,428 @@
+		}
+		if v, ok := computeResource["tags"]; ok {
+			input.ComputeResources.Tags = tagsFromMapGeneric(v.(map[string]interface{}))
+		}
+	}
+
+	if _, err := conn.CreateComputeEnvironment(input); err != nil {
+		return err
+	}
+
+	d.SetId(computeEnvironmentName)
+
+	if err := waitComputeEnvironmentValid(d, meta); err != nil {
+		return err
+	}
+
+	return resourceAwsBatchComputeEnvironmentRead(d, meta)
+}
+
+func resourceAwsBatchComputeEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	computeEnvironmentName := d.Get("compute_environment_name").(string)
+
+	input := &batch.DescribeComputeEnvironmentsInput{
+		ComputeEnvironments: []*string{
+			aws.String(computeEnvironmentName),
+		},
+	}
+
+	result, err := conn.DescribeComputeEnvironments(input)
+	if err != nil {
+		return err
+	}
+
+	if len(result.ComputeEnvironments) == 0 {
+		return fmt.Errorf("One compute environment is expected, but AWS return no compute environment")
+	} else if len(result.ComputeEnvironments) >= 2 {
+		return fmt.Errorf("One compute environment is expected, but AWS return too many compute environment")
+	}
+	computeEnvironment := result.ComputeEnvironments[0]
+
+	d.Set("service_role", computeEnvironment.ServiceRole)
+	d.Set("state", computeEnvironment.State)
+	d.Set("type", computeEnvironment.Type)
+
+	if *(computeEnvironment.Type) == "MANAGED" {
+		computeResource := computeEnvironment.ComputeResources
+
+		d.Set("compute_resources", []interface{}{
+			map[string]interface{}{
+				"bid_percentage":      computeResource.BidPercentage,
+				"desired_vcpus":       computeResource.DesiredvCpus,
+				"ec2_key_pair":        computeResource.Ec2KeyPair,
+				"image_id":            computeResource.ImageId,
+				"instance_role":       computeResource.InstanceRole,
+				"instance_type":       schema.NewSet(schema.HashString, flattenStringList(computeResource.InstanceTypes)),
+				"max_vcpus":           computeResource.MaxvCpus,
+				"min_vcpus":           computeResource.MinvCpus,
+				"security_group_ids":  schema.NewSet(schema.HashString, flattenStringList(computeResource.SecurityGroupIds)),
+				"spot_iam_fleet_role": computeResource.SpotIamFleetRole,
+				"subnets":             schema.NewSet(schema.HashString, flattenStringList(computeResource.Subnets)),
+				"tags":                tagsToMapGeneric(computeResource.Tags),
+				"type":                computeResource.Type,
+			},
+		})
+	}
+
+	d.Set("arn", computeEnvironment.ComputeEnvironmentArn)
+	d.Set("ecc_cluster_arn", computeEnvironment.ComputeEnvironmentArn)
+	d.Set("status", computeEnvironment.Status)
+	d.Set("status_reason", computeEnvironment.StatusReason)
+
+	return nil
+}
+
+func resourceAwsBatchComputeEnvironmentDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	computeEnvironmentName := d.Get("compute_environment_name").(string)
+
+	log.Printf("[DEBUG] Delete compute environment \"%s\".\n", computeEnvironmentName)
```

Comment:
Could you put it line 296, in the same way that we did for create & read, in order to provide better debug?